dotCMS/src/integration-test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentActionletTest.java

```diff
@@ -6,57 +6,179 @@
+        final ContentType contentType = createTestType();
+        contentTypes.add(contentType);
 
         // associated the scheme to the type
         final WorkflowScheme systemWorkflowScheme = workflowAPI.findSystemWorkflowScheme();
-        workflowAPI.saveSchemesForStruct(new StructureTransformer(customContentType).asStructure(),
+        workflowAPI.saveSchemesForStruct(new StructureTransformer(contentType).asStructure(),
                 Collections.singletonList(systemWorkflowScheme));
 
+        final User userWithPermission = createUserWithPermission(contentType);
+        final User userWithoutPermission = new UserDataGen().nextPersisted();
+        final User userWithJustActionPermission = createUserWithJustActionPermission();
+
+        final ContentType frontendContentType = createTestType();
+        contentTypes.add(frontendContentType);
+
+        final User frontEndUserWithPermission = createFrontendUser(frontendContentType);
+        final User frontendUserWithoutPermission = createFrontendUserWithoutPermission();
+
+        return new TestCase[]{
+                new TestCase(true, APILocator.systemUser(), contentType, true, true),
+                new TestCase(false, APILocator.systemUser(), contentType, true, true),
+                new TestCase(true, userWithPermission, contentType, true, true),
+                new TestCase(false, userWithPermission, contentType, true, true),
+                new TestCase(true, userWithoutPermission, contentType, false, false),
+                new TestCase(false, userWithoutPermission, contentType, false, false),
+                new TestCase(true, userWithJustActionPermission, contentType, true, false),
+                new TestCase(false, userWithJustActionPermission, contentType, true, false),
+                new TestCase(true, frontEndUserWithPermission, frontendContentType, true, true),
+                new TestCase(false, frontEndUserWithPermission, frontendContentType, true, true),
+                new TestCase(true, frontendUserWithoutPermission, frontendContentType, false, false),
+                new TestCase(false, frontendUserWithoutPermission, frontendContentType, false, false)
+        };
+    }
+
+    private static User createFrontendUser(final ContentType contentType) {
+
+        try {
+            final Role role = APILocator.getRoleAPI().loadFrontEndUserRole();
+            final User user = new UserDataGen().roles(role).nextPersisted();
+
+            addPermissionToAddChildren(role, contentType);
+            addPermissionToSaveAction(role);
+
+            return user;
+        } catch (DotDataException e) {
+            throw  new RuntimeException(e);
+        }
+    }
+
+    private static User createFrontendUserWithoutPermission() {
+
+        try {
+            final Role role = APILocator.getRoleAPI().loadFrontEndUserRole();
+            final User user = new UserDataGen().roles(role).nextPersisted();
+
+            return user;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider simply returning the value vs storing it in local variable 'user'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)